pkg/plugins/resources/k8s/store.go

```diff
@@ -165,6 +171,20 @@ func (s *KubernetesStore) List(ctx context.Context, rs core_model.ResourceList,
 	return nil
 }
 
+func (s *KubernetesStore) countK8sResources(ctx context.Context, rs core_model.ResourceList) (int, error) {
```

Comment:
The problem with this implementation is that when dealing with a high number of objects (ex. 100 000) you have to load everything into memory just to count it even if the size page is 100.

Talk with @hbagdi since I think he proposed an alternative solution.

Reply:
This is a naive implementation indeed, as explained in the PR description. Adding a controller as proposed might have its advantages, but will complicate the code. I suggest we measure what is the performance with a large number of objects and then decide on optimisations. My feeling is this will suffice with anything below 10,000 resources, which should be fine for now IMO.

Reply:
> with a high number of objects (ex. 100 000)

Realistically speaking, is this possible for any resource in Kubernetes? If so, it is non-trivial to solve the problems that arises with that scale of resources in k8s and should be out of scope of this PR.

I'm not familiar with the current code-base but here is how you can optimize this:
- If you are not already caching the result in the controller, please do so
- You can list the resources once, and then use informers to track create and delete events and change the counts accordingly.